scripts/parse_vcgs_manifest.py

```diff
@@ -0,0 +1,381 @@
+                project=self.sample_metadata_project, new_sample=new_sample
+            )
+            external_sample_id_to_internal_id[new_sample.external_id] = sample_id
+
+        for sample_id, sequences_to_add in sequencing_to_add.items():
+            for seq in sequences_to_add:
+                seq.sample_id = external_sample_id_to_internal_id[sample_id]
+                seqapi.create_new_sequence(
+                    project=self.sample_metadata_project, new_sequence=seq
+                )
+
+        return external_sample_id_to_internal_id
+
+    def parse_sequencing_type(self, sample_id: str, types: List[str]):
+        # filter false-y values
+        types = list(set(t for t in types if t))
+        if len(types) <= 0:
+            return self.default_sequencing_type
+        if len(types) > 1:
+            raise ValueError(
+                f'Multiple library types for same sample {sample_id}, '
+                f'maybe there are multiples types of sequencing in the same '
+                f'manifest? If so, please raise an issue with mfranklin to '
+                f'change the groupby to include {Columns.LIBRARY_STRATEGY}.'
+            )
+
+        type_ = types[0].lower()
+        if type_ == 'wgs':
+            return 'wgs'
+        if type_ in ('single-cell', 'ss'):
+            return 'single-cell'
+
+        raise ValueError(f'Unrecognised sequencing type {type_}')
+
+    def parse_reads(self, reads: List[str]) -> Union[List[List[Dict]], List[Dict]]:
+        if all(r.endswith('.fastq') or r.endswith('.fastq.gz') for r in reads):
+            structured_fastqs = self.parse_fastqs_structure(reads)
+            files = []
+            for fastq_group in structured_fastqs:
+                files.append([self.create_file_object(f) for f in fastq_group])
+
+            return files
+
+        elif all(r.endswith('.cram') for r in reads):
+            sec_format = ['.crai', '^.crai']
+            files = []
+            for r in reads:
+                secondaries = self.create_secondary_file_objects_by_potential_pattern(
+                    r, sec_format
+                )
+                files.append(self.create_file_object(r, secondary_files=secondaries))
+
+            return files
+
+        elif all(r.endswith('.bam') for r in reads):
+            sec_format = ['.bai', '^.bai']
+            files = []
+            for r in reads:
+                secondaries = self.create_secondary_file_objects_by_potential_pattern(
+                    r, sec_format
+                )
+                files.append(self.create_file_object(r, secondary_files=secondaries))
+
+            return files
+
+        extensions = set(os.path.splitext(r)[-1] for r in reads)
+
+        joined_reads = ''.join(f'\n\t{i}: {r}' for i, r in enumerate(reads))
+        raise ValueError(
+            f'Mixed, or unrecognised extensions ({", ".join(extensions)}) for reads:{joined_reads}'
+        )
+
+    def parse_fastqs_structure(self, fastqs):
```

Comment:
Maybe worth adding type hints here. And probably clarify about the return structure. Is it a list of pairs (R1/R2), or a pair of lists (different lanes for the same read orientation)? Realised that I got confused :)